apps/host/src/components/modules/challenges/challenge-grid/challenge-grid.tsx

```diff
@@ -13,13 +13,42 @@ interface Props {
 }
 
 function ChallengeGrid({ challenges, linkPrefix, links }: Props) {
-  const [searchInput, setSearchInput] = useState('');
+  const initialSearchFilters = () => {
+    const filters = sessionStorage.getItem('searchFilters');
+    if (filters) {
+      const parsedFilters = JSON.parse(filters);
+      return {
+        searchInput: parsedFilters.searchInput || '',
+        optionSelected: parsedFilters.optionSelected || [],
+        selectedDifficulties: parsedFilters.selectedDifficulties || [],
+        isSegmentBtn1: parsedFilters.isSegmentBtn1 || false,
+        selectedChallengesByTags: parsedFilters.selectedChallengesByTags || [],
+        newChallenge: parsedFilters.newChallenge || false,
+      };
+    }
+    return {
+      searchInput: '',
+      optionSelected: [],
+      selectedDifficulties: [],
+      isSegmentBtn1: false,
+      selectedChallengesByTags: [],
+      newChallenge: false,
+    };
+  };
```

Comment:
if `initialSearchFilters` has no dependency on the component, can we not move everything to util

Reply:
Okay, I'll try.